client/signup/config/step-components.js

```diff
@@ -85,6 +85,9 @@ export function getStepModuleMap() {
 }
 export async function getStepComponent( stepName ) {
 	const moduleName = stepNameToModuleName[ stepName ];
+	if ( ! moduleName ) {
+		return;
```

Comment:
I'm not too familiar with this design, but I think this should throw an error? Is this an optional or required mapping for all the steps? If required, we should probably throw an error because this would probably result in WSOD if someone tried to load an unmapped step.

Reply:
I'd find it odd for this to throw an error. After all, someone might call this in a non-signup context, and then they should just get `undefined` IMHO. And usually getter functions would return `null` or `undefined` when something is not found. And anyone using a getter function should be double-checking the return.

I do get your concerns, though, that it might result in concealing errors, so I'm happy to trigger a `console.error` instead of throwing an error. 

As a follow-up, we could also add some tracks if we're extra concerned. Furthermore, it would help if this was typed.

Reply:
Logging an error in 150b382fac9db8cf3521999f0589fcbfcf163f30